### ISSUE TITLE
alive does not build with parallel make on base 3.15.5 or 7.0.1.1

### ISSUE THREAD

**Issue by MarkRivers** [epics-modules/alive#5]
alive does not build with parallel make on base 7.0.1.1.  This is the error:
```
corvette:~/devel/alive>make -sj
make[2]: *** No rule to make target `../../lib/linux-x86_64/libalive.a', needed by `aliveTest'.  Stop.
make[2]: *** Waiting for unfinished jobs....
In file included from /corvette/usr/local/epics-devel/base-7.0.1/include/dbAccess.h:19:0,
                 from ../aliveTest.cpp:4:
/corvette/usr/local/epics-devel/base-7.0.1/include/dbBase.h:162:18: warning: 'rset' is deprecated (declared at /corvette/usr/local/epics-devel/base-7.0.1/include/recSup.h:88) [-Wdeprecated-declarations]
     rset        *prset;
                  ^
In file included from /corvette/usr/local/epics-devel/base-7.0.1/include/dbAccess.h:19:0,
                 from ../aliveRecord.c:39:
/corvette/usr/local/epics-devel/base-7.0.1/include/dbBase.h:162:5: warning: 'rset' is deprecated (declared at /corvette/usr/local/epics-devel/base-7.0.1/include/recSup.h:67) [-Wdeprecated-declarations]
     rset        *prset;
     ^
In file included from /corvette/usr/local/epics-devel/base-7.0.1/include/dbAccess.h:23:0,
                 from ../aliveTest.cpp:4:
/corvette/usr/local/epics-devel/base-7.0.1/include/dbAccessDefs.h:207:59: warning: 'rset' is deprecated (declared at /corvette/usr/local/epics-devel/base-7.0.1/include/recSup.h:88) [-Wdeprecated-declarations]
 epicsShareFunc rset * dbGetRset(const struct dbAddr *paddr);
                                                           ^
In file included from /corvette/usr/local/epics-devel/base-7.0.1/include/dbAccess.h:23:0,
                 from ../aliveRecord.c:39:
/corvette/usr/local/epics-devel/base-7.0.1/include/dbAccessDefs.h:207:46: warning: 'rset' is deprecated (declared at /corvette/usr/local/epics-devel/base-7.0.1/include/recSup.h:67) [-Wdeprecated-declarations]
 epicsShareFunc rset * dbGetRset(const struct dbAddr *paddr);
                                              ^
../aliveRecord.c:90:1: warning: 'rset' is deprecated (declared at /corvette/usr/local/epics-devel/base-7.0.1/include/recSup.h:67) [-Wdeprecated-declarations]
 rset aliveRSET =
 ^
../aliveRecord.c:111:1: warning: 'rset' is deprecated (declared at /corvette/usr/local/epics-devel/base-7.0.1/include/recSup.h:67) [-Wdeprecated-declarations]
 epicsExportAddress(rset,aliveRSET);
 ^
../aliveRecord.c:111:1: warning: 'rset' is deprecated (declared at /corvette/usr/local/epics-devel/base-7.0.1/include/recSup.h:67) [-Wdeprecated-declarations]
../aliveRecord.c:111:1: warning: 'rset' is deprecated (declared at /corvette/usr/local/epics-devel/base-7.0.1/include/recSup.h:67) [-Wdeprecated-declarations]
make[1]: *** [install.linux-x86_64] Error 2
make: *** [tests.install] Error 2
make: *** Waiting for unfinished jobs....
```
So it is trying to build tests before aliveLib is built.  I don't understand this because it looks like the tests_DEPEND_DIRS is correctly defined in the [Makefile](https://github.com/epics-modules/alive/blob/master/Makefile)

I tried changing the Makefile to remove the tests_DEPEND_DIRS from the BASE_3_15 conditional, and also to explicitly make it aliveApp like this:
```
#Tests uses dbUnitTests.h from base-3.15
ifdef BASE_3_15
DIRS := $(DIRS) $(filter-out $(DIRS), tests)
endif

tests_DEPEND_DIRS += aliveApp
```
But that did not help.

**Comment by MarkRivers** [comment]
I just found that this problem also occurs on base 3.15.5.

```
corvette:~/support/alive>make -sj clean uninstall
corvette:~/support/alive>make -sj
make[2]: *** No rule to make target `../../lib/linux-x86_64/libalive.a', needed by `aliveTest'.  Stop.
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [install.linux-x86_64] Error 2
make: *** [tests.install] Error 2
make: *** Waiting for unfinished jobs....
```

**Comment by MarkRivers** [comment]
I found and fixed the problem.  The following change is needed.  Without that it resets the tests_DEPEND_DIRS to only include configure, rather than configure and aliveApp.

```
corvette:~/devel/alive>git diff Makefile
diff --git a/Makefile b/Makefile
index b89379e..d867256 100644
--- a/Makefile
+++ b/Makefile
@@ -13,7 +13,7 @@ endif


 define DIR_template
- $(1)_DEPEND_DIRS = configure
+ $(1)_DEPEND_DIRS += configure
 endef
 $(foreach dir, $(filter-out configure,$(DIRS)),$(eval $(call DIR_template,$(dir))))
```

**Comment by MarkRivers** [comment]
Closed via ae95174766e44fa94626a8d7eea37332ebbd717c.
